pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -93,11 +94,23 @@ func (k *klusterletStatusController) sync(ctx context.Context, controllerContext
 	workDesiredCondition := checkAgentDeploymentDesired(ctx, k.kubeClient, agentNamespace, workDeploymentName, klusterletWorkDesiredDegraded)
 	workDesiredCondition.ObservedGeneration = klusterlet.Generation
 
-	_, _, err = helpers.UpdateKlusterletStatus(ctx, k.klusterletClient, klusterletName,
+	updateFuncs := []helpers.UpdateKlusterletStatusFunc{
 		helpers.UpdateKlusterletConditionFn(availableCondition),
 		helpers.UpdateKlusterletConditionFn(registrationDesiredCondition),
 		helpers.UpdateKlusterletConditionFn(workDesiredCondition),
-	)
+	}
+
+	if klusterlet.Spec.HubApiServerHostAlias != nil {
```

Comment:
I think the major problem is klusterlet will use hubKubeConfig to talk to hub with sar request to check connectivity.
When hostAliase is set, klusterlet should also use this hostAliase to talk to hub.

Reply:
I have a little bit confused.   

When i set the hostAliase into klusterlet deployment, the klusterlet_ssar_controller will use this hostAliase to talk to hub. And the status of klusterlet is updated as shown below. I think this is working.

```yaml
status:
  conditions:
  - lastTransitionTime: "2022-07-03T07:42:27Z"
    message: Hub kubeconfig secret open-cluster-management-agent/hub-kubeconfig-secret
      to apiserver https://xxx.yyy.zzz is working
    observedGeneration: 14
    reason: HubConnectionFunctional
    status: "False"
    type: HubConnectionDegraded
```

Reply:
do you want do it as a followup or in this one? Could be a follow up I think.